app/src/main/java/org/mozilla/fenix/library/bookmarks/selectfolder/SelectBookmarkFolderAdapter.kt

```diff
@@ -96,8 +98,36 @@ class SelectBookmarkFolderAdapter(private val sharedViewModel: BookmarksSharedVi
             .orEmpty()
     }
 
+    private fun getSelectedItemIndex(): Int? {
+        val selectedNode = sharedViewModel.selectedFolder
+        val selectedNodeIndex = currentList.indexOfFirst { it.node == selectedNode }
+
+        return selectedNodeIndex.takeIf { it != -1 }
+    }
+
+    @Suppress("NOTHING_TO_INLINE")
+    private inline fun BookmarkNode.isSelected(): Boolean =
+        this == sharedViewModel.selectedFolder
+
     companion object {
         private const val maxDepth = 10
         private const val dpsToIndent = 10
     }
 }
+
+private class DiffCallback : DiffUtil.ItemCallback<BookmarkNodeWithDepth>() {
+
+    override fun areItemsTheSame(
+        oldItem: BookmarkNodeWithDepth,
+        newItem: BookmarkNodeWithDepth
+    ) = (oldItem.node.guid == newItem.node.guid)
```

Comment:
nit: remove parenthesis 
```suggestion
    ) = oldItem.node.guid == newItem.node.guid
```